### DISCUSSION TITLE
Can't inject to server-side

### DISCUSSION THREAD

**Discussion by OutsiderH** [FabricMC/fabric-api#3564] [Mod Dev Support] [answered]
### Description

When I try to inject the method that in the class without `@Environment` Annotation, it not work on both integrated and dedicated server.
I can only use `src\main\java\modid\mixin` to inject the method that in the class with `@Environment(value=EnvType.SERVER)` Annotation. And `src\client\java\modid\mixin\client` to inject `@Environment(value=EnvType.CLIENT)`.
My fabric.mod.json is like
```
"mixins": [
	"modid.mixins.json",
	{
		"config": "modid.client.mixins.json",
		"environment": "client"
	}
],
```
And modid.mixins.json
```
{
	"required": true,
	"package": "modid.mixin",
	"compatibilityLevel": "JAVA_17",
	"mixin": [
		"ClassName"
	],
	"injectors": {
		"defaultRequire": 1
	}
}
```

**Comment by Linguardium** [comment]
What you have currently is common side (single player and dedicated server). is that what you wanted? You should not be using `@Environment` annotation at all

**Reply by OutsiderH** [reply]
I am not using this annotation, I just want to inject the class without this annotation like ServerPlayerEntity

**Reply by OutsiderH** [reply]
If i type some code like this
```
@Mixin(ServerPlayerEntity.class)
public abstract class ServerPlayerEntityFix {
	@Inject(method = "<init>(Lnet/minecraft/server/MinecraftServer;Lnet/minecraft/server/world/ServerWorld;Lcom/mojang/authlib/GameProfile;Lnet/minecraft/network/encryption/PlayerPublicKey;)V", at = @At("TAIL"))
	private void post_con(CallbackInfo info) {
		throw new NotImplementedException();
	}
}
```
in common side then is not work on both single and mutiplayer, if i type that in client side instead, it work on singleplayer.
So I am looking a way to inject the class like that(without `@Environment`) to the dedicated server

**Comment by OutsiderH** [comment, marked as answer]
I am using "mixin" instead of "mixins" in modid.mixins.json
